datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -283,10 +300,12 @@ protected void sendDatastreamProducerRecord(DatastreamProducerRecord datastreamP
       TopicPartition srcTopicPartition, int numBytes, SendCallback sendCallback) {
     _producer.send(datastreamProducerRecord, ((metadata, exception) -> {
       if (exception != null) {
-        String msg = String.format("Detect exception being thrown from callback for src partition: %s while "
-            + "sending, metadata: %s , exception: ", srcTopicPartition, metadata);
+        String msg = String.format("Detected exception being thrown from send callback for source topic-partition: %s "
+            + "with metadata: %s, exception: ", srcTopicPartition, metadata);
         _logger.warn(msg, exception);
-        rewindAndPausePartitionOnException(srcTopicPartition, exception);
+        synchronized (_sendFailureTopicPartitionExceptionMap) {
+          _sendFailureTopicPartitionExceptionMap.put(srcTopicPartition, exception);
```

Comment:
You can extract this out into another function and reuse between both connectorTasks

Reply:
done